code/game/objects/items/emags.dm

```diff
@@ -64,8 +65,21 @@
 	return ITEM_INTERACT_SUCCESS
 
 /obj/item/card/emag/ranged_interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	if (can_emag(interacting_with, user))
+		if(pshoom_2 && prob(1))
+			playsound(src, pshoom_2, 40, TRUE)
+		else
+			playsound(src, pshoom, 40, TRUE)
+		user.Beam(interacting_with, icon_state = "rped_upgrade", time = 0.5 SECONDS)
 	return prox_check ? NONE : interact_with_atom(interacting_with, user)
 
+/obj/item/card/emag/item_interaction(mob/living/user, obj/item/tool)
+	if (istype(tool, /obj/item/assembly/signaler/anomaly/bluespace))
+		qdel(src)
```

Comment:
Move qdel(src) to the end--I think that once a thing is qdel'd nothing should be doing anything with it anymore.

Reply:
if i qdel(src) at the end, it'll try to place an object in the active hand before deleting the src item (the emag), no?